specs/vep-2448-vdk-run-logs-simplified-and-readable/README.md

```diff
@@ -0,0 +1,251 @@
+The VDK user experience when it comes to using logs for data job runs is lacking
+in several areas. This makes it hard to debug failed data job runs and get
+useful information out of successful data job runs. The following sections
+provide an in-depth view of improvemnts to VDK run logs which address a number
+of user pain points, such as noisy logs, overly verbose error messages,
+insufficient error information, lack of documentation, lack of progress tracking
+features, etc. Resolving these pain points will result in a better user
+experience.
+
+## Glossary
+<!--
+Optional section which defines terms and abbreviations used in the rest of the document.
+-->
+
+## Motivation
+<!--
+It tells **why** do we need X?
+Describe why the change is important and the benefits to users.
+Explain the user problem that need to be solved.
+-->
+
+At the moment, VKD users face multiple challenges related to logging in data job
+runs. Based on user feedback we've received, these can be split into 5 categories.
+
+### Logs on failures
+
+Useful information gets lost in the noise or sometimes just
+missing This is especially true for user errors. There are also some error
+messages that just don't contain any useful information to begin with We might
+have incentivized some bad practices with our error message format. Some people
+tend to just write placeholder text, e.g. "WHAT HAPPENED: An error occurred", or
+"COUNTERMEASURES: Check exception". These kinds of messages are extremely
+annoying from an user perspective. When a step fails, we don't point to the
+actual line of the code where the failure happened Actual root causes are
+sometimes buried in the stack There's lots of meta-logs, impala query logs, etc.
+This leads us to believe that we have a lot of info logging that should actually
+be on the debug level.
+
+### Logs on success
+
+Users expect little or no log output on success. Users tend to add their own
+logging to signify when a step started or ended they expect these kinds of logs
+to be highlighted somehow, or be the only logs they see
+
+### Debug mode
+
+Some users have trouble running jobs in debug mode and getting debug level logs
+This points to a gap in our documentation
+
+### Progress Tracking
+
+Users want better tracking of job progress (for DAGs, steps, ingestion)
+
+### VDK exception handling and  categorization
+
+There is a need for more clarity and categorization in error messages, likely
+more structured error reporting, such as knowing whether an error is a syntax
+error, data error, library issue, or platform error, as very important
+visibility of the steps or components where failures occur so people can quickly
+find out the principal issue
+
+Logging and error handling should be improved in ways that adress all these
+challenges. This will result in less time spent troubleshooting, fewer support
+queries, more data job runs that are successful and an overall better user
+experience.
+
+## Requirements and goals
```

Comment:
Now that I think about it, I am not sure we are 100% ready for the requirements section. It's something we write after we have some good idea on the solution and finalize the scope only. 

So let's revisit it later after the solution research?

Reply:
VEPs should be improved in iterations, so we can absolutely revisit it. I expect our requirements to evolve as we do more research and brainstorming.